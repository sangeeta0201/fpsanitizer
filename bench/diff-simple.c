#include <stdio.h>
#include <math.h>
#include <string.h>

int main() {
  double x,y;
  x = 1e16;
  y = (x + 1) - x;
  printf("y:%e\n", y);
  return 0;
}
