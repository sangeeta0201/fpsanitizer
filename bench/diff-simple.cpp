#include <iostream>
#include <stdio.h>
#include <math.h>
#include <string.h>

void check(){
  std::string fileName = "sangeeta";
  std::cout<<fileName<<"\n"; 
}
int main() {
  double x,y;
  x = 1e16;
  y = (x + 1) - x;
  printf("%e\n", y);
//  size_t result_A = (size_t) &y;
 // printErrorD(result_A, y);
//  printf("Result using float:%e\n", y);  
  return 0;
}
