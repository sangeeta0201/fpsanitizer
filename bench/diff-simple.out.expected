(output
  (function "main")
  (filename "diff-simple.c")
  (line-num 8)
  (instr-addr 400580)
  (avg-error 61.998590)
  (max-error 61.998590)
  (num-calls 1)
  (influences
    (
    (
     (expr
       (FPCore ()
          (- (+ 1.000000 1.000000e16) 1.000000e16)))
     (var-problematic-ranges)
     (example problematic input ())
     (function "main")
     (filename "diff-simple.c")
     (line-num 7)
     (instr-addr 40055B)
     (avg-error 61.998590)
     (max-error 61.998590)
     (avg-local-error 61.998590)
     (max-local-error 61.998590)
     (num-calls 1))
    )
  )
)
